base/shell/explorer/startmnusite.cpp

```diff
@@ -112,6 +115,56 @@ class CStartMenuSite :
         return FALSE;
     }
 
+    void FillDocumentsMenu(HMENU hDocMenu)
+    {
+        WCHAR szUser[128];
+        DWORD cchUser = _countof(szUser);
+        GetUserNameW(szUser, &cchUser);
+
+        MENUITEMINFOW info = { sizeof(info), MIIM_TYPE };
+        info.fType = MFT_STRING;
+        info.dwTypeData = szUser;
+        SetMenuItemInfo(hDocMenu, 0, TRUE, &info);
+
+        while (DeleteMenu(hDocMenu, 4, MF_BYPOSITION))
+        {
+            ;
+        }
+
+        WCHAR szDir[MAX_PATH], szSpec[MAX_PATH];
+        SHGetSpecialFolderPathW(NULL, szDir, CSIDL_RECENT, FALSE);
```

Comment:
Is there no api for this?

Reply:
I don't know yet.

Reply:
`comctl32!CreateMRUListW`
`HKCU\SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\RecentDocs`